### DISCUSSION TITLE
GitLab SSH Handshake failed: EOF

### DISCUSSION THREAD

**Discussion by andreasvikke** [fluxcd/flux2#2536] [Q&A] [answered]
After changing the a Github repo to Gitlab repo, i've run into the following error:
`unable to clone 'ssh://git@gitlab.com/xxxxx/xxxxx.git': ssh: handshake failed: EOF`

The error occurs at random times, sometimes at night and others mid day. When it happens it sends around 2-10 messages on slack and then nothing for hours/days.

Flux is installed trough Terraform
`provider registry.terraform.io/fluxcd/flux v0.11.0`

Flux check output:
```bash
❯ flux check
► checking prerequisites
✗ flux 0.17.2 <0.27.3 (new version is available, please upgrade)
✔ kubectl 1.22.1 >=1.18.0-0
✔ Kubernetes 1.20.12-gke.1500 >=1.16.0-0
► checking controllers
✔ helm-controller: deployment ready
► ghcr.io/fluxcd/helm-controller:v0.17.0
✔ image-automation-controller: deployment ready
► ghcr.io/fluxcd/image-automation-controller:v0.20.0
✔ image-reflector-controller: deployment ready
► ghcr.io/fluxcd/image-reflector-controller:v0.16.0
✔ kustomize-controller: deployment ready
► ghcr.io/fluxcd/kustomize-controller:v0.21.0
✔ notification-controller: deployment ready
► ghcr.io/fluxcd/notification-controller:v0.22.0
✔ source-controller: deployment ready
► ghcr.io/fluxcd/source-controller:v0.21.2
✔ all checks passed
```

**Comment by darkowlzz** [comment, marked as answer]
Hi, I've seen this exact issue when using GitLab. It seems like an issue on the gitlab server itself. 
It has helped to increase the `spec.interval` for the git repositories to reduce the frequency of this failure. But it still happens once in a while.
I've been working on sending a positive notification after such failures to signal that everything is alright in https://github.com/fluxcd/source-controller/pull/604 . I hope that'll help in knowing about the failure better.

**Reply by andreasvikke** [reply]
Thanks for the answer, i will up the `spec.interval` for now.

If you get a solution merged, i would be glad to hear from you!

**Reply by gertux** [reply]
Do you happen to know what the root cause on the GitLab server might be ?

We're seeing the same issue and would like to fix it on the GitLab server.

**Reply by joshsizer** [reply]
Seeing the same thing as well, maybe 2 or 3 times a day

**Reply by tomaaron** [reply]
We're seeing the same thing as well, a couple of times a day all of a sudden.
